### PR TITLE
Add OCP/OKD requirements to Backup/Restore test and code cleanup

### PR DESCRIPTION
**Pull request by dmvolod**

### CODE REVIEW

File: test/e2e/backup_restore_test.go [hunk 1]
```diff
@@ -53,6 +53,10 @@ func testBackupRestore(clusterSpec clusterSpec) func(*testing.T) {
 
 		// 3. Backup the cluster's content
 		backupSpec := &v2.Backup{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "infinispan.org/v2alpha1",
+				Kind:       "Backup",
+			},
```
Comment (ryanemerson) [inline test/e2e/backup_restore_test.go:59]: Why is adding the `TypeMeta` necessary?

File: pkg/controller/zerocapacity/zero_controller.go [hunk 2]
```diff
@@ -217,31 +216,23 @@ func (z *Controller) initializeResources(request reconcile.Request, instance Res
 	}
 
 	if err := ensureClusterStability(infinispan); err != nil {
-		return reconcile.Result{}, fmt.Errorf("Infinispan not stable: %w", err)
-	}
-
-	statefulset := &appsv1.StatefulSet{}
```
Comment (ryanemerson) [inline pkg/controller/zerocapacity/zero_controller.go:223]: Good spot. I forgot to remove this when I updated how the zero pod was configured.

File: test/e2e/backup_restore_test.go (hunk 1 above)
Comment (dmvolod) [reply test/e2e/backup_restore_test.go:59]: GVK requires on OCP/OKD for Update. Probably not in this case, but we should don't forget about it
Please have a look at this case and code
https://github.com/infinispan/infinispan-operator/blob/f44347dfd4c5484425a75b11ddde07f0b5fda941/test/e2e/main_test.go#L706-L708

File: pkg/controller/zerocapacity/zero_controller.go [hunk 3]
```diff
@@ -217,31 +216,23 @@ func (z *Controller) initializeResources(request reconcile.Request, instance Res
...
-	if err := z.Client.Get(ctx, clusterKey, statefulset); err != nil {
-		z.Log.Info(fmt.Sprintf("Unable to load Infinispan StatefulSet '%s': %w", clusterName, err))
-		if errors.IsNotFound(err) {
-			return reconcile.Result{RequeueAfter: consts.DefaultWaitOnCluster}, nil
-		}
-		return reconcile.Result{}, err
+		z.Log.Info("Infinispan not ready: %s", err.Error())
```
[the remark this answers is not in the document]
Comment (ryanemerson) [reply pkg/controller/zerocapacity/zero_controller.go:219]: LGTM

File: pkg/apis/infinispan/v1/types_util.go [hunk 4]
```diff
@@ -248,6 +248,17 @@ func (ispn *Infinispan) NotClusterFormed(pods, replicas int) bool {
 	return notFormed || notEnoughMembers
 }
 
+func (ispn *Infinispan) EnsureClusterStability() error {
```
Comment (ryanemerson) [inline pkg/apis/infinispan/v1/types_util.go:251]: I don't think `EnsureClusterStability` belongs in `type_util` as the logic is specific to the zero controller and it's perception of whether the cluster is stable enough for the zero pod to be created.

Comment (dmvolod) [reply pkg/apis/infinispan/v1/types_util.go:251]: @ryanemerson if you can see, I switched IsWellFormed function to use EnsureClusterStability as you implementation more correct for cluster validation than just check single condition

Comment (ryanemerson) [reply pkg/apis/infinispan/v1/types_util.go:251]: O right, I missed that. In the case let's keep it. 

`IsWellFormed` is only used by the cache controller, maybe we should remove `IsWellFormed` and just update the call to:  
```go
if err := ispnInstance.EnsureClusterStability(); err != nil {
...
}
```

Comment (dmvolod) [reply pkg/apis/infinispan/v1/types_util.go:251]: I've found IsWellFormed in `autoscale` and test also. Are you agree to change all possible places with `EnsureClusterStability`. If yes, I will do.

Comment (ryanemerson) [reply pkg/apis/infinispan/v1/types_util.go:251]: Sorry my search was on an outdated `master` .. if it has a few uses I don't mind keeping it, but whichever you prefer :+1:

File: pkg/controller/zerocapacity/zero_controller.go [hunk 5]
```diff
@@ -419,17 +398,14 @@ func (z *Controller) zeroPodSpec(name, namespace string, configMap *corev1.Confi
 	}
 
 	if zeroSpec.Volume.UpdatePermissions {
-		pod.Spec.InitContainers = []corev1.Container{
-			ispnCtrl.ChmodInitContainer("backup-chmod-pv", name, zeroSpec.Volume.MountPath),
-		}
+		ispnCtrl.AddVolumeChmodInitContainer("backup-chmod-pv", name, zeroSpec.Volume.MountPath, &pod.Spec)
 	}
 
-	ispnCtrl.AddServerDataChmodInitContainer(dataVolName, &pod.Spec)
```
Comment (dmvolod) [inline pkg/controller/zerocapacity/zero_controller.go:427]: @ryanemerson do we really need to add chmod init container for zero pod data volume as it's EmptyDir, not real volume
Also OCP tests were fails as backup-chmod-pv adds without lookup MAKE_DATADIR_WRITABLE env variable.
wdyt about this?

Comment (ryanemerson) [reply pkg/controller/zerocapacity/zero_controller.go:427]: > do we really need to add chmod init container for zero pod data volume as it's EmptyDir, not real volume

Possibly not. I made this change when I was investigating the CI failures, so I think we can remove it.



> Also OCP tests were fails as backup-chmod-pv adds without lookup MAKE_DATADIR_WRITABLE env variable.
> wdyt about this?

Right, makes sense to check the env var then. I think minikube will fail without the initContainer, but I can just set `MAKE_DATADIR_WRITABLE` in that case.

Comment (dmvolod) [reply pkg/controller/zerocapacity/zero_controller.go:427]: Thanks, already fixed.

**Comment by ryanemerson** [comment]
Thanks @dmvolod

**Pull request merged**